sdk/servicebus/azure-servicebus/azure/servicebus/_pyamqp/aio/_session_async.py

```diff
@@ -372,10 +372,10 @@ async def _incoming_detach(self, frame):
         try:
             link = self._input_handles[frame[0]]  # handle
             await link._incoming_detach(frame)  # pylint: disable=protected-access
-            # if link._is_closed:  TODO
-            #     self.links.pop(link.name, None)
-            #     self._input_handles.pop(link.remote_handle, None)
-            #     self._output_handles.pop(link.handle, None)
+            if link._is_closed:  # TODO - need for this
```

Comment:
Added this in for detaching after a drain -> we need to clear the handle otherwise we get an error